services/distribution/src/main/java/app/coronawarn/server/services/distribution/config/DistributionServiceConfig.java

```diff
@@ -654,4 +660,129 @@ public void setMinAndroid(String minAndroid) {
+      return iosKeyDownloadParameters;
+    }
+
+    public void setIosKeyDownloadParameters(IosKeyDownloadParameters iosKeyDownloadParameters) {
+      this.iosKeyDownloadParameters = iosKeyDownloadParameters;
+    }
+
+    public AndroidKeyDownloadParameters getAndroidKeyDownloadParameters() {
+      return androidKeyDownloadParameters;
+    }
+
+    public void setAndroidKeyDownloadParameters(AndroidKeyDownloadParameters androidKeyDownloadParameters) {
+      this.androidKeyDownloadParameters = androidKeyDownloadParameters;
+    }
+
+    public IosExposureDetectionParameters getIosExposureDetectionParameters() {
+      return iosExposureDetectionParameters;
+    }
+
+    public void setIosExposureDetectionParameters(IosExposureDetectionParameters iosExposureDetectionParameters) {
+      this.iosExposureDetectionParameters = iosExposureDetectionParameters;
+    }
+
+    public AndroidExposureDetectionParameters getAndroidExposureDetectionParameters() {
+      return androidExposureDetectionParameters;
+    }
+
+    public void setAndroidExposureDetectionParameters(
+        AndroidExposureDetectionParameters androidExposureDetectionParameters) {
+      this.androidExposureDetectionParameters = androidExposureDetectionParameters;
+    }
+
+    public static class AndroidKeyDownloadParameters extends CommonKeyDownloadParameters {
+
+      private Integer downloadTimeoutInSeconds;
+      private Integer overallTimeoutInSeconds;
+
+      public Integer getDownloadTimeoutInSeconds() {
+        return downloadTimeoutInSeconds;
+      }
+
+      public void setDownloadTimeoutInSeconds(Integer downloadTimeoutInSeconds) {
+        this.downloadTimeoutInSeconds = downloadTimeoutInSeconds;
+      }
+
+      public Integer getOverallTimeoutInSeconds() {
+        return overallTimeoutInSeconds;
+      }
+
+      public void setOverallTimeoutInSeconds(Integer overallTimeoutInSeconds) {
+        this.overallTimeoutInSeconds = overallTimeoutInSeconds;
+      }
+    }
+
+    private abstract static class CommonKeyDownloadParameters {
+
+      private String cachedDayPackagesToUpdateOnETagMismatch;
+      private String cachedHourPackagesToUpdateOnETagMismatch;
+
+      public String getCachedDayPackagesToUpdateOnETagMismatch() {
+        return cachedDayPackagesToUpdateOnETagMismatch;
+      }
+
+      public void setCachedDayPackagesToUpdateOnETagMismatch(String cachedDayPackagesToUpdateOnETagMismatch) {
+        this.cachedDayPackagesToUpdateOnETagMismatch = cachedDayPackagesToUpdateOnETagMismatch;
+      }
+
+      public String getCachedHourPackagesToUpdateOnETagMismatch() {
+        return cachedHourPackagesToUpdateOnETagMismatch;
+      }
+
+      public void setCachedHourPackagesToUpdateOnETagMismatch(String cachedHourPackagesToUpdateOnETagMismatch) {
+        this.cachedHourPackagesToUpdateOnETagMismatch = cachedHourPackagesToUpdateOnETagMismatch;
+      }
+    }
+
+    public static class IosKeyDownloadParameters extends CommonKeyDownloadParameters {
```

Comment:
Does IosKeyDownloadParameters have no specific params other than the common ones?

Reply:
Nope. Would you suggest to use only the common ones and remove the ios?

Reply:
Not necesarily. I think it's ok to leave it like this as it makes it explicit. I was just wondering whether some fields were forgotton